src/array/ops/broadcast.rs

```diff
@@ -101,9 +112,101 @@ impl BinaryArray {
     }
 }
 
+impl Broadcastable for ListArray {
+    fn broadcast(&self, num: usize) -> DaftResult<Self> {
+        if self.len() != 1 {
+            return Err(crate::error::DaftError::ValueError(format!(
+                "Attempting to broadcast non-unit length Array named: {}",
+                self.name()
+            )));
+        }
+        let maybe_val = self.downcast().iter().next().unwrap();
+        match maybe_val {
+            Some(val) => {
+                let repeated_values = arrow2::compute::concatenate::concatenate(
+                    std::iter::repeat(val.as_ref())
+                        .take(num)
+                        .collect::<Vec<&dyn arrow2::array::Array>>()
+                        .as_slice(),
+                )?;
+                match i64::try_from(val.len()) {
+                    Ok(array_length) => {
+                        let offsets = arrow2::offset::OffsetsBuffer::<i64>::try_from(
+                            std::iter::repeat(array_length)
+                                .take(num)
+                                .fold(vec![0], |mut acc, x| {
+                                    acc.push(x + acc.last().unwrap());
+                                    acc
+                                }),
+                        )?;
+                        if let DataType::List(field) = self.data_type() {
+                            let arrow_arr = Box::new(arrow2::array::ListArray::<i64>::new(
+                                    arrow2::datatypes::DataType::LargeList(Box::new(field.to_arrow()?)),
+                                    offsets,
+                                    repeated_values,
+                                    None,
+                                )
+                            );
+                            ListArray::new(
+                                self.field.clone(),
+                                arrow_arr,
+                            )
+                        } else {
+                            unreachable!("should never be reached");
```

Comment:
Should we propagate up an Error here instead with what happened?

Reply:
This is just to make the compiler happy, there's no way for this branch to be reached at runtime since we're guaranteed that the `datatype()` of `ListArray` is `DataType::List()`. This _should_ be represented in our type system  and I have some ideas for how to do that, but opted to take some prototyping of that out of this PR.

Not sure if there's a better runtime pattern for this kind of thing?

Reply:
Opted for using a `match` and making the `unreachable!()` message nicer.